components/BidsTable.tsx

```diff
@@ -53,9 +56,14 @@ export default function BidsTable(props: Props) {
           {props.bids.map((bid, index) => (
             <tr
               key={bid.tx}
-              onMouseEnter={() => setAddress(bid.walletAddress)}
+              onMouseEnter={() => setAddress(lookup[bid.walletAddress]?.ens || bid.walletAddress)}
               onMouseLeave={() => setAddress('')}
-              className={address === bid.walletAddress ? 'hovered' : ''}
+              className={
+                address === bid.walletAddress ||
+                (lookup[bid.walletAddress]?.ens && address === lookup[bid.walletAddress]?.ens)
```

Comment:
Instead, pass down `winnerAddress` to `<AuctionHeader>` component and only set atom to the winner address, not ENS. This way the `hoveredAddress` atom will always contain an address and you won't need to deal with ENS here